angular/src/lib/select/select.component.ts

```diff
@@ -489,7 +456,29 @@ export class SelectComponent implements AfterContentChecked, AfterContentInit, C
     }, this.closeDuration);
   }
 
-  selectItem(event, option) {
+  private forceSelectItem(emit: boolean = false) {
+    if (this.selectOptionsToDisplay && this.selectOptionsToDisplay.length > 0) {
+      this.onModelChange(this.value);
+      this.finalOption = this.selectedOption;
+      if (emit) {
+        this.handleChange.emit({
+          value: this.value
+        });
+      }
+
+      if (this.tableRowIndex >= 0) {
+        this.tableService.onSelectChange(this.value, this.tableRowIndex);
+      }
+
+      if (this.isMulti) {
+        this.toggleRowWithCheckbox(this.selectedOption.value);
+      } else if (emit) {
+        this.close();
```

Comment:
This close was tripping up the last unit test case which tests clicking the button on the select component and then checking if the overlay is visible.  It wasn't happening because the `click` method under the covers calls `setValue` which then comes in here via `writeValue` which subsequently closes the overlay.  After looking at this, I think we want both the `onSelChange` and the `toggleRowWithCheckBox` to be called because they are setting component state, we just don't want the action of `close` in the case of single-select.

Where this could crop up is sort of a race condition where the user clicks the select at the same time that we are updating it via patchValue or setValue.  In this case, I believe the select component will reflect the change in the value of the control but not close the overlay unexpectedly on the user.